### PR TITLE
Feature/2

### PR DESCRIPTION
**Pull request by Kennygunderman**
Add user auth functionality

### CODE REVIEW

**Comment by Kennygunderman** [comment]
LGTM!

File: lib/main.dart [hunk 1]
```diff
@@ -1,31 +1,77 @@
...
+            create: (_) => authService
+        ),
+        ChangeNotifierProvider<LoginViewModel>(
+            create: (_) => LoginViewModel(authService: authService)
+        ),
+        ChangeNotifierProvider<SignUpViewModel>(
+            create: (_) => SignUpViewModel(authService: authService)
+        ),
+        StreamProvider(
+          create: (context) => context.read<AuthService>().authStateChanges,
```
Comment (Kilo-Loco) [inline lib/main.dart:39]: Couldn't you have used `authService.authStateChanges` here to maintain consistency with the precious providers?

File: lib/service/auth/auth_state.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+enum AuthStatus {
```
Comment (Kilo-Loco) [inline lib/service/auth/auth_state.dart:1]: I've learned that it is now preferred to have enums lowercased in Dart

File: lib/ui/auth/login.dart [hunk 3]
```diff
@@ -1,106 +1,120 @@
...
-            SizedBox(height: 16),
-            TextField(
-              obscureText: true,
-              decoration: InputDecoration(
-                labelStyle: TextStyle(color: Colors.grey.shade100),
-                border: OutlineInputBorder(),
-                labelText: 'Password',
+  Widget _authScreen(BuildContext context, LoginViewModel viewModel) {
+    if (viewModel.state.authStatus == AuthStatus.ERROR) {
+      Future.delayed(Duration.zero, () async {
```
Comment (Kilo-Loco) [inline lib/ui/auth/login.dart:34]: Is this a hack to get the AuthDialog to show? Seems like a delay of zero wouldn't have a huge benefit

**Review by Kilo-Loco — COMMENTED** [review]
This looks really good. Great job! Had a few nit pic comments but that's it 🚀

File: lib/main.dart (hunk 1 above)
Comment (Kennygunderman) [reply lib/main.dart:39]: Ah, yes! Good catch

File: lib/ui/auth/login.dart (hunk 3 above)
Comment (Kennygunderman) [reply lib/ui/auth/login.dart:34]: Yeah, this is a weird workaround that I found. I'm not sure why, but If I present the dialog before returning a widget I get an error. This allows for the widget to be computed before showing the dialog. There is probably a better way to do this though 🤷 
unnamed

**Pull request merged**